eng/scripts/validate-docs-package.ps1

```diff
@@ -0,0 +1,54 @@
+<#
+.SYNOPSIS
+Validates packages by simulating docs CI steps
+
+.PARAMETER Package
+An object describing a package configuration to validate in the format used by
+docs CI onboarding format.
+
+Required attributes: 
+  - name: The name of the package
+
+  Supported optional attributes:
+  - registry: The registry to use for the package if it's not NPM
+
+Example: 
+
+```
+@{
+  name = "@azure/attestation@dev";
+  folder = "./types";
+  registry = "<url>";
+  ...
+}
+```
+
+.PARAMETER WorkingDirectory
+Supplied to `npm install ... --prefix $WorkingDirectory`. The place where the 
+node_modules folder is created and the package is installed
+#>
+
+param(
+  [object] $Package,
+  [string] $WorkingDirectory
+)
+."$PSScriptRoot\..\common\scripts\common.ps1"
+
+function GetResult($success, $package, $output) { 
+  return @{ Success = $success; Package = $package; Output = $output }
+}
+
+$additionalParameters = @()
+if ($Package.registry) {
+  Write-Host $Package.registry
+  $additionalParameters += @("--registry", $Package.registry)
+}
+
+Write-Host "npm install $($Package.name) --prefix $WorkingDirectory $additionalParameters"
+$installOutput = npm install $Package.name --prefix $WorkingDirectory @additionalParameters 2>&1
```

Comment:
I was concerned about parallel access of the same folder within the `node_modules` folder within the temp directory.

After discussion, it definitely does break. My recommendation is to make the following change.

Instead of

```
tempDirectory/
   node-modules/
      pkg_for_install_1/
      pkg_for_install_2/
```

Just split up the node modules so the packages installing don't run into each other:

```
tempDirectory/
   pkg1/
      node_modules/
         pkg_for_install_1/
   pkg2/
      node_modules/
         pkg_for_install_1/
         pkg_for_install_2/

```

Reply:
👍